### PR TITLE
DL-1749 Business details page changes

### PR DESCRIPTION
**Pull request by jmountain-hmrc**
# DL-1749 Business details page changes

**New feature/page changes**

* Split the business details page into up to 4 pages, each asking smaller sections of questions/answers.
* Refactored backend submission logic to handle the new pages

## Checklist

 - [x]  I've made every effort to commit high quality, clean code and I have executed relevant static analyses to be sure
 - [x]  I've included appropriate tests with any code I've added (Unit, Integration, Acceptance etc.)
 - [x]  I've executed the acceptance test pack locally to ensure there are no functional regressions
 - [x]  I've added my code using logical, atomic commits, squashing as appropriate - including the JIRA issue number in the commit message
 - [x]  I've run a dependency check to ensure all dependencies are up to date

### CODE REVIEW

File: app/services/IndexService.scala [hunk 1]
```diff
@@ -434,19 +435,16 @@ class IndexService @Inject()(dataCacheService: Save4LaterService,
   }
 
   // Can remove this when we are confident that no old applications have an empty newAWBusiness (should be after 28 days)
```
Comment (davidmthornton) [inline app/services/IndexService.scala]: Is this comment still required?

File: test/controllers/AlreadyStartingTradingControllerTest.scala [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+        val res = alreadyStartingTradingController.showBusinessDetails(false)
+          .apply(SessionBuilder.buildRequestWithSession(userId, businessType))
+
+        status(res) shouldBe 200
+      }
+    }
+  }
+
+  "save" should {
+    "save the trading name details" when {
```
Comment (davidmthornton) [inline test/controllers/AlreadyStartingTradingControllerTest.scala]: Should this be "save the already started trading details"?

File: test/services/ApplicationServiceTest.scala [hunk 3]
```diff
@@ -63,7 +63,8 @@ class ApplicationServiceTest extends AwrsUnitTestTraits
   def cachedData(legalEntity: BusinessType = testLegalEntity) =
     CacheMap(testUtr, Map("legalEntity" -> Json.toJson(legalEntity),
       "businessCustomerDetails" -> Json.toJson(testReviewDetails),
-      businessDetailsName -> Json.toJson(testBusinessDetails()),
+      businessNameDetailsName -> Json.toJson(testBusinessNameDetails()),
+      tradingStartDetailsName -> Json.toJson(newAWBusiness(proposedStartDate = None)),//Some(TupleDate("01", "04", "2019")))),
```
Comment (davidmthornton) [inline test/services/ApplicationServiceTest.scala]: Can you remove the comment please?

File: test/services/apis/AwrsAPI5Test.scala [hunk 4]
```diff
@@ -77,11 +77,11 @@ class AwrsAPI5Test extends AwrsUnitTestTraits
 
       val result = testAPI5.saveReturnedApplication(feModel, TestUtil.defaultAuthRetrieval)
```
Comment (davidmthornton) [inline test/services/apis/AwrsAPI5Test.scala]: Header doesn't match expected text on this file. Can you use sbt scalastyle to check them?

File: app/controllers/auth/AuthFunctionality.scala [hunk 5]
```diff
@@ -51,9 +51,6 @@ trait AuthFunctionality extends AuthorisedFunctions {
     case er: AuthorisationException =>
       Logger.warn(s"[recoverAuthorisedCalls] Auth exception: $er")
       Unauthorized(views.html.unauthorised())
-    case er                         =>
```
[the remark this answers is not in the document]
Comment (jmountain-hmrc) [reply app/controllers/auth/AuthFunctionality.scala]: They're triggered but I don't think they should be at all - they should throw traditionally and get caught with the error handler.

**Comment by jmountain-hmrc** [comment]
> There are no view tests for the new views added. I appreciate that there aren't any for existing views but aren't we supposed to add them for new with a view to adding them for current views at some point?

I'm of the opinion since it's not really dynamic content, it's already adequately tested enough., with page functionality through the controller test. Viewing a section's details or the index page are, in my view, the places to create a view test since they are dynamic content.

**Comment by davidmthornton** [comment]
I take your point about dynamic content being more important, but if we don't test the static content on pages the only thing protecting us from breaking it is a code review so it could more easily be missed. 

If there are unit tests to say what should be visible on the page we are protecting ourselves in case we remove something by mistake. Otherwise the only way we will know if the views are right is by stepping through the journey manually.

Take Trading Name view as an example. There is untested logic in there. If I start removing "stuff" from that view, the controller tests still pass because they aren't checking any of the logic in the view. 

If that logic is shared across multiple views then perhaps we need helpers which are shared and then test them instead?

**Comment by jmountain-hmrc** [comment]
I've added a couple of simple view tests for pages with a bit of dynamic content, after some reflection and overlooking into their functionality

**Pull request merged**